src/components/UserSelector.tsx

```diff
@@ -1,6 +1,54 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { User } from '../types/User';
+
+type Props = {
+  setShowError: (val: boolean) => void,
+  setTrgetUserId: (num: number) => void,
+};
+
+export const UserSelector: React.FC<Props> = ({
+  setShowError,
+  setTrgetUserId,
+}) => {
+  const [showUsers, setShowUsers] = useState<boolean>(false);
+  const [users, setUsers] = useState<User[]>([]);
+  const [targetUser, setTargetUser] = useState<User | null>(null);
+
+  async function getUser() {
+    const BASE_URL = 'https://mate.academy/students-api/users';
+
+    const allUsers = await fetch(BASE_URL)
+      .then(resp => resp.json());
```

Comment:
be consistent use async/await or .then but not both

Reply:
sorry....it was  my experement))) I forgot to remove.